pkg/armhelpers/mockclients.go

```diff
@@ -412,33 +548,10 @@ func (mc *MockAKSEngineClient) ListVirtualMachines(ctx context.Context, resource
 			},
 		},
 	}
-
-	vmr := compute.VirtualMachineListResult{}
-	vmr.Value = &[]compute.VirtualMachine{vm1}
-
-	return &MockVirtualMachineListResultPage{
-		Fn: func(lastResults compute.VirtualMachineListResult) (compute.VirtualMachineListResult, error) {
-			return compute.VirtualMachineListResult{}, nil
-		},
-		Vmlr: vmr,
-	}, nil
-}
-
-//ListVirtualMachineScaleSets mock
-func (mc *MockAKSEngineClient) ListVirtualMachineScaleSets(ctx context.Context, resourceGroup string) (compute.VirtualMachineScaleSetListResultPage, error) {
-	if mc.FailListVirtualMachineScaleSets {
-		return compute.VirtualMachineScaleSetListResultPage{}, errors.New("ListVirtualMachines failed")
-	}
-
-	return compute.VirtualMachineScaleSetListResultPage{}, nil
 }
 
-//GetVirtualMachine mock
-func (mc *MockAKSEngineClient) GetVirtualMachine(ctx context.Context, resourceGroup, name string) (compute.VirtualMachine, error) {
-	if mc.FailGetVirtualMachine {
-		return compute.VirtualMachine{}, errors.New("GetVirtualMachine failed")
-	}
-
+//MakeFakeVirtualMachine returns a fake compute.VirtualMachine
+func (mc *MockAKSEngineClient) MakeFakeVirtualMachine() compute.VirtualMachine {
```

Comment:
nit: since this is already a mock function, using "fake" in the name is redundant. Also slight preference for mock over fake in comment wording.

Reply:
as I already commented before, the mockClient stubs functions like `CreateVirtualMachine` that are real function on the service. This is a helper builder function that return a default instance of a VirtualMachine. the `Fake` part of the name is to distinguish it from the functions exposed by the actual service that we mock

Reply:
So you're saying existing `MockAKSEngineClient` methods (e.g., `ListVirtualMachineScaleSets`) represent mock equivalents of identically names methods in the real client libraries?

If that's the case, I'd argue we don't actually want to be adding these "fake" methods onto pre-existing mock libraries that fit that description, and better to create static functions that access the mock objects in order to return an appropriate fake, single-use object for testing.

I don't want to block progress, but something to think about as the mock + fake semantics are arguably a bit confusing.

Reply:
yes, I extracted this for reusability as it was inlined in the code and duplicated. I first wanted it standalone, but it relied on other parameters from the mockClient and so, I decided to keep it there for simplicity. it is used by the client to generate the faked results, and it can be used from the tests to build dummy objects.